addons/binding/org.openhab.binding.keba/src/main/java/org/openhab/binding/keba/handler/KeContactBroadcastListener.java

```diff
@@ -0,0 +1,196 @@
+            .getScheduledPool(THING_HANDLER_THREADPOOL_NAME);
+    private Set<KeContactHandler> listeners = Collections.synchronizedSet(new HashSet<KeContactHandler>());
+
+    final Logger logger = LoggerFactory.getLogger(KeContactBroadcastListener.class);
+
+    public void start() {
+
+        if (!isStarted) {
+            logger.info("Starting the Keba broadcast listener");
+            try {
+                broadcastChannel = DatagramChannel.open();
+                broadcastChannel.socket().bind(new InetSocketAddress(LISTENER_PORT_NUMBER));
+                broadcastChannel.configureBlocking(false);
+
+                try {
+                    broadcastListener = Selector.open();
+                } catch (IOException e) {
+                    logger.error("An exception occurred while registering the selector: '{}'", e.getMessage());
+                }
+
+                logger.info("Listening for incoming data on {}", broadcastChannel.getLocalAddress());
+
+                synchronized (broadcastListener) {
+                    broadcastListener.wakeup();
+                    try {
+                        broadcastKey = broadcastChannel.register(broadcastListener, broadcastChannel.validOps());
+                    } catch (ClosedChannelException e1) {
+                        logger.error("An exception occurred while registering a selector: {}", e1.getMessage());
+                    }
+                }
+
+                if (broadcastJob == null || broadcastJob.isCancelled()) {
+                    broadcastJob = scheduler.scheduleWithFixedDelay(broadcastRunnable, 0, LISTENING_INTERVAL,
+                            TimeUnit.MILLISECONDS);
+                }
+
+                isStarted = true;
+
+            } catch (Exception e) {
+                logger.error("An exception occurred while creating the broadcast channel on port number {} ({})",
+                        LISTENER_PORT_NUMBER, e.getMessage(), e);
+            }
+        }
+    }
+
+    public void stop() {
+        if (isStarted) {
+            logger.info("Stopping the Keba broadcast listener");
+            if (broadcastJob != null && !broadcastJob.isCancelled()) {
+                broadcastJob.cancel(true);
+                broadcastJob = null;
+            }
+
+            try {
+                broadcastChannel.close();
+            } catch (IOException e) {
+                logger.error("An exception occurred while closing the broadcast channel on port number {} ({})",
+                        LISTENER_PORT_NUMBER, e.getMessage());
+            }
+
+            isStarted = false;
+        }
+    }
+
+    public void registerHandler(KeContactHandler handler) {
+
+        if (handler != null) {
+            listeners.add(handler);
+
+            if (logger.isTraceEnabled()) {
+                logger.trace("There are now {} Keba handlers registered with the broadcast listener", listeners.size());
+            }
+
+            if (listeners.size() == 1) {
```

Comment:
How is that guaranteed, what is the use of line 119 then. Why is it a set then and not a normal field or an optional?